tests/vm_delete_protection_test.go

```diff
@@ -30,7 +30,7 @@ var _ = Describe("VM delete protection", func() {
 			err := apiClient.Get(ctx, client.ObjectKeyFromObject(vm), vm)
 			Expect(err).To(Or(Not(HaveOccurred()), MatchError(errors.IsNotFound, "errors.IsNotFound")))
 
-			if err == nil {
+			if err == nil && vm.DeletionTimestamp != nil {
```

Comment:
If `vm.DeletionTimestamp != nil`, that means that the VM is being deleted, and we should not call `Update`.